### DISCUSSION TITLE
v24.2

### DISCUSSION THREAD

**Discussion by alex-w** [Stellarium/stellarium#3778] [General]
**The installers are located in the "Assets" section at the end of these release notes!**

[![Github Release](https://img.shields.io/github/downloads/Stellarium/stellarium/v24.2/total.svg)](https://github.com/Stellarium/stellarium/releases/v24.2)

## Release notes
The Stellarium Team has released version 24.2.

The major changes of this version:
- Improvements in plugins and AstroCalc tools
- Updates in sky cultures
- Updates in GUI

Behind the scenes, many more minor issues were fixed.

TELESCOPE USERS

We have identified and fixed some bugs in the TelescopeControl plugin on
the Qt6-based releases.

Telescope users, please try whether this solves your problems.
If not, we are looking for those of you with programming skills to help
us fixing remaining issues. Reward possible!

## What's Changed
Full list of changes:
- Added ability to show an approximate limiting magnitude of devices in Oculars plug-in
- Added support Twilight Factor and Relative Brightness for binoculars in Oculars plug-in
- Added support Adler Index for binoculars in Oculars plug-in
- Added support Bishop Index for binoculars in Oculars plug-in
- Added more actions for Exoplanets plug-in
- Added set of navigational stars from Nevil Maskelyne's "The British Mariner's Guide" published in 1764
- Added discovery circumstances for newly discovered comets
- Added partial solar eclipse opening angle
- Added ability to follow CMAKE_INSTALL_MANDIR variable (GH: #3732)
- Added ability to add/subtract great years via hotkeys
- Added support for localization for solar eclipses maps
- Added export of an equirectangular raster eclipse map
- Added missing discovery circumstances for some comets
- Added note for unmarked entries for Sources in Satellites plugin (GH: #3617)
- Fixed eclipse artifacts (GH: #3729)
- Fixed building Scenery3D plugin with Qt 6.7 (GH: #3763)
- Fixed Russian translation of Belarusian SC description
- Fixed image path in German translation of Greek (Farnese) SC
- Fixed references in Romanian SC
- Fixed Galician translation of Xhosa SC description
- Fixed Italian translation of Anutan SC description
- Fixed calculation of twilight factor (or dusk index) for binoculars in Oculars plug-in
- Fixed hiding/restoring of star name display in Navigational Stars plugin (GH: #3728)
- Fixed morphological description of DSO
- Fixed output for CLI option --list-landscapes in Windows
- Fixed solar radius
- Fixed CLI output in Windows (GH: #3710)
- Fixed generator of KML 
- Fixed properties for some periodic comets
- Fixed remembering the sorting rules between updates in AstroCalc/Positions tool (GH: #3684)
- Fixed failure to parse too small numbers with from_chars
- Changed AstroCalc/Eclipses tool: check number of penumbra limits in terms of rise-set line
- Changed AstroCalc/Eclipses tool: rewrite generation of shadow limits
- Changed AstroCalc/Eclipses tool: rewrite AstroCalcDialog::getMaximumEclipseAtRiseSet
- Changed AstroCalc/Eclipses tool: rewrite AstroCalcDialog::getRiseSetLineCoordinates
- Changed core/AstroCalc: Move solar eclipse computations out from Planet.cpp
- Changed core/AstroCalc: Move some solar eclipse computations from AstroCalcDialog
- Changed core/AstroCalc: do not show earthshine during annular or high-percentage partial eclipses
- Changed core/AstroCalc: refine max eclipse at rise/set curve
- Changed core: force redraw of NomenclatureItems if needed (GH: #3706)
- Changed core: rename sky culture change signal for better consistency
- Changed core: switch off IAU moon numbers from screen labels
- Changed Satellites plug-in: compactification of GUI
- Changed Satellites plug-in: improve explanatory tooltip
- Changed Remote Control plug-in: add and fix missing functions and features
- Changed scaling limits, special for real planetariums
- Changed GUI: minor clarification
- Changed Satellites plugin: expanding photometry and RCS data for satellites 
- Changed Scenery3D plugin: rename variable for debugging of the shaders (GH: #3670)
- Changed visual style: enable table borders by default in sky cultures
- Changed visual style: use single lines between table cells in sky cultures
- Updated Vanuatu (Netwar) sky culture: clarified the license with author
- Updated Anutan sky culture: fix formatting
- Updated Chinese sky culture: fix formatting
- Updated Tibetian sky culture: rename an image to match the name in the links
- Updated default catalog of exoplanets
- Updated default list of locations
- Updated default catalog of pulsars
- Updated discovery circumstances data for minor planets
- Updated discovery circumstances data for comets
- Updated photometry of satellites
- Updated planetary nomenclature
- Updated group for RPM
- Removed extraneous borders in tables in About dialogs of plugins (GH: #3742)
- Removed config option for an AppImage (GH: #3726)

## Packages notes
package | note
------------|---------
stellarium-24.2.tar.gz | Stellarium 24.1 for UNIX/Linux (source code). <br />__MD5:__ 27a7cd3ab9421cb73790db6022f7c5aa <br />__SHA256:__ e6d8ee0792b7f77486b700d4669d0dd0c349319f379758ad643d76165d1d56d2
stellarium-24.2.tar.bz2 | Stellarium 24.1 for UNIX/Linux (source code). <br />__MD5:__ ca705a368a3598561eb1d25df70e7676 <br />__SHA256:__ 55d55d4983c7263a32c23a579389283171dd9bb42e783cb26882a9546e85a7c5
stellarium-24.2.tar.xz | Stellarium 24.1 for UNIX/Linux (source code). <br />__MD5:__ 97ebab4102e455ab73672d9caca1fbbe <br />__SHA256:__ f9e8e114e34265194ffff0c2a2806a6fdc998449192749c4ef60b0a5302101af
stellarium-24.2.zip | Stellarium 24.1 for UNIX/Linux (source code). <br />__MD5:__ b46261ae4b205f06246b59957a9bf572 <br />__SHA256:__ 47e7702d90b1da5db4062a3c967924aedd071e424c070cc7c3a32ed6ca927f03
stellarium-24.2-qt5-win32.exe | Stellarium 24.1 for Windows (x86_32; Windows 7+) based on Qt 5.12.6 (Visual Studio 2017). <br /> __MD5__: a85e5a7e7a3d8d41ec0177152171e959 <br />__SHA256__: 0f7963b69f0acb8192b91ab12e15f2ccda953ed39c0c3c47e34bfd53f6ce02a8
stellarium-24.2-qt5-win64.exe | Stellarium 24.1 for Windows (x86_64; Windows 7+) based on Qt 5.12.6 (Visual Studio 2017). <br /> __MD5__: fc98ea2a43f4a1bacb689995c83d0322 <br />__SHA256__: cb0241677b878167f1c2b45477b0c4a3dd11bf9c2806e950193480de7632ebfa
stellarium-24.2-qt6-win64.exe | Stellarium 24.1 for Windows (x86_64; Windows 10+) based on Qt 6.5.3 (Visual Studio 2019). <br /> __MD5__: 9bb771c93364092286f0bb062ce90bc4 <br />__SHA256__: d9db848d9d11597c2a916f0ababf5c489103714834706ab1c79f1b374c954cb2
stellarium-24.2-qt6-arm64.exe | Stellarium 24.1 for Windows (ARM64; Windows 10+) based on Qt 6.5.3 (Visual Studio 2022). <br /> __MD5__: 554c7900cc2d624b5be67a77048c3c30 <br />__SHA256__: 7e445b9d213e305b67f6e004ebe2e463cee6df5a6515b82adf79ee9db95e9a54
Stellarium-24.2-qt6-macOS.zip | Stellarium 24.1 for macOS (universal; macOS 11.0+) based on Qt 6.5.3 (AppleClang 15.0.0). <br /> __MD5__: 29571975d74c4c23c86bd7cfcbe3a6b7 <br />__SHA256__: 34bdec53ba5dfb2569a52545ab06306f12843ef483970d548c59cd49e77b5c99
Stellarium-24.2-qt5-x86_64.zip | Stellarium 24.1 for macOS (x86_64; macOS 10.14+) based on Qt 5.12.12 (AppleClang 13.0.0). <br /> __MD5__: 5808774430dbd3a5ab9e299510357ef4 <br />__SHA256__: cd8c12db299dfb20fcc80f608bf81f4c913535676cd37c062067bd5875952eef
Stellarium-24.2-qt5-x86_64.AppImage | Stellarium 24.1 for Linux (amd64; glibc 2.31) based on Qt 5.12.8 (GCC 9.4.0). <br /> __MD5__: a7a4c75ed5e5ab4a2df287ae76e23e56 <br />__SHA256__: a619c1e2f28920b9b77a0a19ee5baf17907934b4a45596d64ab16e43b3a1547d
Stellarium-24.2-qt6-x86_64.AppImage | Stellarium 24.1 for Linux (amd64; glibc 2.35) based on Qt 6.2.4 (GCC 11.4.0). <br /> __MD5__: 1a32f601687092c9d615df7c366384be <br />__SHA256__: b463602f09d48f02f9e2813d5e74faea690d83db63dae63db84e55755c3d26d3

### GPG signature
1. Import Stellarium's GPG key
  `curl -O https://stellarium.org/files/stellarium.gpg`
2. List Stellarium's GPG key
  `gpg --with-fingerprint --show-keys --keyid-format long stellarium.gpg`
  **id:** `rsa4096/BF38D4D02A328DFF 2012-04-07`
  **fingerprint:** `7915 1C2E 6351 E727 8DA1  A730 BF38 D4D0 2A32 8DFF`
3. Verify the binary file is valid (tarball for example)
  `gpgv --keyring ./stellarium.gpg stellarium-24.2.tar.gz.asc stellarium-24.2.tar.gz`

<hr /><em>This discussion was created from the release <a href='https://github.com/Stellarium/stellarium/releases/tag/v24.2'>v24.2</a>.</em>

**Comment by 10110111** [comment]
> We have identified and fixed some bugs in the TelescopeControl plugin on
the Qt6-based releases.

We haven't fixed them. All the tentative fixes appeared to be ineffective.

**Reply by gzotti** [reply]
Indeed. We still need practical users of the Telescope Plugin who are impeded enough by this to become active...

**Reply by 10110111** [reply]
Given the existing Qt5 workaround, I guess we'll not find any. Actually, the problem is now easily reproducible in the VM, it's just that it's quite nontrivial to diagnose. May even be a bug in Qt6, but again, this is hard to diagnose.

**Reply by gzotti** [reply]
Yes, it's a spiral - as long as we release Qt5 builds, it may not be solved. But I feel the Telescope plugin has many demands open. From supporting AltAz mounts (required modifying the ancient protocol) over updating INDI version to this family of issues with ASCOM driven telescopes which as you say is hard to solve. Computer-driven astronomy used to be a field for computer experts who developed much by themselves. OK, now it's mainstream, and we cannot expect the average user to be able to understand what's under the hood. :-(

At some point, Qt5 will go away, and the raze of Win10-only systems next year will likely remove the need for Angle or --low-graphics. Or maybe contrary - cheap old Win10-only laptops will flood the "I bought that cheap W10 notebook for outdoor/offline use" market.
